Source/JavaScriptCore/runtime/WeakMapPrototype.cpp

```diff
@@ -85,9 +87,9 @@ JSC_DEFINE_HOST_FUNCTION(protoFuncWeakMapGet, (JSGlobalObject* globalObject, Cal
     if (!map)
         return JSValue::encode(jsUndefined());
     JSValue key = callFrame->argument(0);
-    if (!key.isObject())
+    if (UNLIKELY(!canBeHeldWeakly(key)))
         return JSValue::encode(jsUndefined());
```

Comment:
ditto as above, is it even worth the isRegistered check here since it'll usually be false and just cost us extra loads

Reply:
Yeah, changed it to `isCell()` check.